packages/flutter/test/widgets/system_context_menu_test.dart

```diff
@@ -895,4 +895,400 @@ void main() {
+                        contextMenuBuilder:
+                            (BuildContext context, EditableTextState editableTextState) {
+                              return SystemContextMenu.editableText(
+                                editableTextState: editableTextState,
+                                items: <IOSSystemContextMenuItem>[
+                                  IOSSystemContextMenuItemCustom(
+                                    title: 'Field 2 Action',
+                                    onPressed: () {
+                                      field2ActionCalled = true;
+                                    },
+                                  ),
+                                ],
+                              );
+                            },
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+      );
+      
+      await tester.longPress(find.byType(TextField).first);
+      await tester.pump();
+      expect(find.byType(SystemContextMenu), findsOneWidget);
+      
+      final EditableTextState state1 = tester.state<EditableTextState>(find.byType(EditableText).first);
+      final SystemContextMenu menu1 = tester.widget<SystemContextMenu>(find.byType(SystemContextMenu));
+      final IOSSystemContextMenuItemCustom item1 = menu1.items[0] as IOSSystemContextMenuItemCustom;
+      
+      ByteData message = const StandardMethodCodec().encodeMethodCall(
+        MethodCall('SystemContextMenu.onPerformCustomAction', item1.hashCode.toString()),
+      );
+      await TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.handlePlatformMessage(
+        'flutter/platform',
+        message,
+        (_) {},
+      );
+      
+      expect(field1ActionCalled, isTrue);
+      expect(field2ActionCalled, isFalse);
+      
+      state1.hideToolbar();
+      await tester.pump();
+      
+      field1ActionCalled = false;
+      
+      await tester.longPress(find.byType(TextField).last);
+      await tester.pump();
+      expect(find.byType(SystemContextMenu), findsOneWidget);
+      
+      final EditableTextState state2 = tester.state<EditableTextState>(find.byType(EditableText).last);
+      final SystemContextMenu menu2 = tester.widget<SystemContextMenu>(find.byType(SystemContextMenu));
+      final IOSSystemContextMenuItemCustom item2 = menu2.items[0] as IOSSystemContextMenuItemCustom;
+      
+      message = const StandardMethodCodec().encodeMethodCall(
+        MethodCall('SystemContextMenu.onPerformCustomAction', item2.hashCode.toString()),
+      );
+      await TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.handlePlatformMessage(
+        'flutter/platform',
+        message,
+        (_) {},
+      );
+      
+      expect(field1ActionCalled, isFalse);
+      expect(field2ActionCalled, isTrue);
+      
+      state2.hideToolbar();
+      await tester.pump();
```

Comment:
Nit: Maybe do a final expect that SystemContextMenu is not found. Ending on a `pump` makes me wonder what was being tested there.